ignition_ros2_control/src/ignition_ros2_control_plugin.cpp

```diff
@@ -0,0 +1,477 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <controller_manager/controller_manager.hpp>
+
+#include <hardware_interface/resource_manager.hpp>
+#include <hardware_interface/component_parser.hpp>
+#include <hardware_interface/types/hardware_interface_type_values.hpp>
+
+#include <ignition/gazebo/components/Joint.hh>
+#include <ignition/gazebo/components/JointType.hh>
+#include <ignition/gazebo/components/Name.hh>
+#include <ignition/gazebo/components/ParentEntity.hh>
+#include <ignition/gazebo/components/World.hh>
+#include <ignition/gazebo/Model.hh>
+
+#include <ignition/plugin/Register.hh>
+
+#include <pluginlib/class_loader.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "ignition_ros2_control/ignition_ros2_control_plugin.hpp"
+#include "ignition_ros2_control/ignition_system.hpp"
+
+namespace ignition_ros2_control
+{
+//////////////////////////////////////////////////
+class IgnitionROS2ControlPluginPrivate
+{
+public:
+  /// \brief Get the URDF XML from the parameter server
+  std::string getURDF() const;
+
+  /// \brief Get a list of enabled, unique, 1-axis joints of the model. If no
+  /// joint names are specified in the plugin configuration, all valid 1-axis
+  /// joints are returned
+  /// \param[in] _entity Entity of the model that the plugin is being
+  /// configured for
+  /// \param[in] _ecm Ignition Entity Component Manager
+  /// \return List of entities containinig all enabled joints
+  std::map<std::string, ignition::gazebo::Entity> GetEnabledJoints(
+    const ignition::gazebo::Entity & _entity,
+    ignition::gazebo::EntityComponentManager & _ecm) const;
+
+  /// \brief Entity ID for sensor within Gazebo.
+  ignition::gazebo::Entity entity_;
+
+  /// \brief Node Handles
+  std::shared_ptr<rclcpp::Node> node;
```

Comment:
Nit to match the rest

```suggestion
  std::shared_ptr<rclcpp::Node> node_;
```